### PR TITLE
Use nftables to configure firewall

### LINKED ISSUE #1112: Add support of Ubuntu 22

### PR DESCRIPTION
**Pull request by badrogger**
**Changes**
- Use nftables to configure firewall. 
- Do not open unnecessary ports
- Use golang based docker-compose binary. 

**Performance**
- No performance related changes was introduced.

**Testing**
- Tested on local network.
- Added unit tests. 

The following should be performed to verify the changes:
- node creation
- update
- chain creation
- chain deletion
- turn-off/turn-on
- backup/restore
-  rebooting the server

### CODE REVIEW

File: node_cli/core/nftables.py [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+
+
+try:
+    import nftables
+except (FileNotFoundError, AttributeError, ModuleNotFoundError) as err:
+    if "pytest" in sys.modules or ENV == 'dev':
+        from collections import namedtuple  # hotfix for tests
+        iptc = namedtuple('nftables', ['Chain', 'Rule'])
+    else:
+        logger.error(f'Unable to import iptc due to an error {err}')
```
Comment (dmytrotkk) [inline node_cli/core/nftables.py]: should be `Unable to import nftables...` instead of `iptc`

File: node_cli/operations/__init__.py [hunk 2]
```diff
@@ -25,5 +25,6 @@
     turn_off as turn_off_op,
     turn_on as turn_on_op,
     restore as restore_op,
-    repair_sync as repair_sync_op
+    repair_sync as repair_sync_op,
+    configure_nftables as configure_firewall
```
Comment (dmytrotkk) [inline node_cli/operations/__init__.py]: what is the point of exporting this function under a different name here?

it is used 6 times as `configure_nftables` and 1 time as `configure_firewall` - suggesting keeping just one version for consistency.

File: node_cli/core/node.py [hunk 3]
```diff
@@ -294,13 +259,16 @@ def update(env_filepath: str, pull_config_for_schain: str, unsafe_ok: bool = Fal
         error_msg = 'Cannot update safely'
         error_exit(error_msg, exit_code=CLIExitCodes.UNSAFE_UPDATE)
 
+    prev_version = get_meta_info().version
+    logger.info('HERE %s %s', __version__, prev_version)
+    if (__version__ == 'test' or __version__.startswith('2.6')) and prev_version == '2.5.0':
```
Comment (dmytrotkk) [inline node_cli/core/node.py:268]: will it work ok if `migrate` will be executed multiple times? (let's say for 2.6.1, 2.6.2, etc)

File: node_cli/migrations/focal_to_jammy.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
+import logging
```
Comment (dmytrotkk) [inline node_cli/migrations/focal_to_jammy.py:20]: no license header

File: node_cli/utils/helper.py [hunk 5]
```diff
@@ -414,3 +414,35 @@ def get_tmp_path(path: str) -> str:
...
+
+def get_ssh_port(ssh_service_name='ssh'):
+    try:
+        return socket.getservbyname(ssh_service_name)
+    except OSError:
+        logger.exception('Cannot get ssh service port')
+        return DEFAULT_SSH_PORT
+
+
+def remove_between_brackets(text: str, pattern: str) -> str:
```
Comment (dmytrotkk) [inline node_cli/utils/helper.py:427]: confusing name - maybe let's use a small docstring with a function explanation?

**Review by dmytrotkk — CHANGES_REQUESTED** [review]

**Review by dmytrotkk — APPROVED** [review]

**Review by DmytroNazarenko — APPROVED** [review]

File: node_cli/core/node.py [hunk 6]
```diff
@@ -98,7 +98,7 @@ class NodeStatuses(Enum):
 
 
 def is_update_safe(sync_node: bool = False) -> bool:
-    if not sync_node and (not is_admin_running() or not is_api_running()):
+    if not sync_node and (not is_admin_running() and not is_api_running()):
```
Comment (dmytrotkk) [inline node_cli/core/node.py]: after the latest fix, redundant braces can be removed:

```python
if not sync_node and not is_admin_running() and not is_api_running():
```

**Review by dmytrotkk — APPROVED** [review]

**Pull request merged**